### PR TITLE
update NServiceBus step by step tutorial to v8.x and net6.0

### PR DESCRIPTION
**Pull request by soujay**
Update NSB step by step tutorial to NSB 8 and net6.0

### CODE REVIEW

**Comment by ramonsmits** [comment]
@soujay is it intentional that you removed the v7 variant?

**Comment by DavidBoike** [comment]
@ramonsmits Yes, tutorials should not teach you how to use NServiceBus with an old version.

File: tutorials/nservicebus-step-by-step/1-getting-started/tutorial.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ In this first lesson, which should take 10-15 minutes, you will create your firs
 
 Let's build something simple to give NServiceBus a try.
 
-Although [NServiceBus only requires .NET Framework 4.5.2](/nservicebus/operations/dotnet-framework-version-requirements.md), this tutorial assumes at least [Visual Studio 2017](https://www.visualstudio.com/downloads/) and .NET Framework 4.6.1.
+This tutorial uses the latest version of NServiceBus and it requires [.NET 6 (or higher) or .NET Framework Version 4.7.2 (or higher)](/nservicebus/operations/dotnet-framework-version-requirements.md)  and  assumes at least [Visual Studio 2022](https://www.visualstudio.com/downloads/) 
 
 NOTE: You can also build the solution in .NET Core by selecting the **Console App (.NET Core)** project type when you create a new project.
```
Comment (DavidBoike) [inline tutorials/nservicebus-step-by-step/1-getting-started/tutorial.md:26]:
```suggestion
This tutorial uses NServiceBus version 8, .NET 6, and assumes an up-to-date installation of [Visual Studio 2022](https://www.visualstudio.com/downloads/).

NOTE: NServiceBus 8 also [supports .NET Framework 4.7.2 or higher](/nservicebus/operations/dotnet-framework-version-requirements.md), but [new applications should be built on .NET 6 or higher](https://devblogs.microsoft.com/dotnet/net-core-is-the-future-of-net/). 
```

File: tutorials/nservicebus-step-by-step/3-multiple-endpoints/Snippets/Core_8/Lesson3.cs [hunk 2]
```diff
@@ -12,7 +12,7 @@ public async Task SendTypes(IEndpointInstance endpointInstance, IMessageHandlerC
 
             #region SendLocal
             // From endpoint startup code
-            await endpointInstance.SendLocal(command)
+            await endpointInstance.SendLocal(command, context.CancellationToken)
```
Comment (DavidBoike) [inline tutorials/nservicebus-step-by-step/3-multiple-endpoints/Snippets/Core_8/Lesson3.cs:15]: These are included together for ease of writing the docs, but the token from the message handler context shouldn't be passed to the endpointInstance methods. Revert this and suppress the warning using an attribute on the SendTypes method. Set the justification in the attribute to be "Parameters only both included for ease of documentation"

File: tutorials/nservicebus-step-by-step/3-multiple-endpoints/Snippets/Core_8/Lesson3.cs [hunk 3]
```diff
@@ -22,7 +22,7 @@ await context.SendLocal(command)
 
             #region Send
             // From endpoint startup code
-            await endpointInstance.Send(command)
+            await endpointInstance.Send(command, context.CancellationToken)
```
Comment (DavidBoike) [inline tutorials/nservicebus-step-by-step/3-multiple-endpoints/Snippets/Core_8/Lesson3.cs:25]: Here too

File: tutorials/nservicebus-step-by-step/3-multiple-endpoints/Snippets/Core_8/Lesson3.cs [hunk 4]
```diff
@@ -32,7 +32,7 @@ await context.Send(command)
 
             #region SendDestination
             // Not recommended, most of the time
-            await endpointInstance.Send("Destination", command)
+            await endpointInstance.Send("Destination", command, context.CancellationToken)
```
Comment (DavidBoike) [inline tutorials/nservicebus-step-by-step/3-multiple-endpoints/Snippets/Core_8/Lesson3.cs:35]: And here

**Review by DavidBoike — CHANGES_REQUESTED** [review]
Some more changes (outside the current set, so I can't do a line comment) are also needed:

* [Lesson 1 Create a solution](https://docs.particular.net/tutorials/nservicebus-step-by-step/1-getting-started/#exercise-create-a-solution) change to assume net6 or higher.
  * We don't have to talk about CompletedTask API anymore. The minimum frameworks supported by NServiceBus 8 already have it.
* [Lesson 1 Configure an endpoint](https://docs.particular.net/tutorials/nservicebus-step-by-step/1-getting-started/#exercise-configure-an-endpoint) take out the [paragraph about C# 7.1 features](https://docs.particular.net/tutorials/nservicebus-step-by-step/1-getting-started/#exercise-configure-an-endpoint) - with net6.0 you get them for free.
  * [This section](https://github.com/Particular/docs.particular.net/blob/master/tutorials/nservicebus-step-by-step/1-getting-started/tutorial.md?plain=1#L63-L69) should not mention C# 7.1, should not have a `PreCsharp7-1AsyncBoilerplate` snippet,rename the `AsyncMain` snippet to `Main`, and rewrite the sentence before (line 67) so that it only talks about the Main method.
  * Look other places for "AsyncMain" (in all 5 lessons) and clean that up as well.

**Review by DavidBoike — APPROVED** [review]
Just one nitpick

**Pull request merged**